src/nvim/tui/tui.c

```diff
@@ -189,12 +191,47 @@ TUIData *tui_start(int *width, int *height, char **term)
                  100, 0);
 
   loop_poll_events(&main_loop, 1);
-  *width = tui->width;
-  *height = tui->height;
-  *term = tui->term;
   return tui;
 }
 
+void tui_attach(TUIData *tui)
+{
+  MAXSIZE_TEMP_ARRAY(args, 3);
+  ADD_C(args, INTEGER_OBJ(tui->width));
+  ADD_C(args, INTEGER_OBJ(tui->height));
+
+  MAXSIZE_TEMP_DICT(opts, 9);
+  PUT_C(opts, "rgb", BOOLEAN_OBJ(true));
+  PUT_C(opts, "ext_linegrid", BOOLEAN_OBJ(true));
+  PUT_C(opts, "ext_termcolors", BOOLEAN_OBJ(true));
+  if (tui->term) {
+    PUT_C(opts, "term_name", STRING_OBJ(cstr_as_string(tui->term)));
```

Comment:
It should be `PUT_C` and `cstr_as_string()` here to avoid leaking memory. When this was in `ui_client_run()` ASAN didn't complain about this because `ui_client_run()` never returns, so the allocated memory was still reachable.